tcl/cpu/arc/v3_32.tcl

```diff
@@ -0,0 +1,257 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+#  Copyright (C) 2022 Synopsys, Inc.
+#  Shahab Vahedi <shahab@synopsys.com>
+
+source [find cpu/arc/common.tcl]
+
+# Currently 'examine_target' can only read JTAG registers and set properties -
+# but it shouldn't write any of registers - writes will be cached, but cache
+# will be invalidated before flushing after examine_target, and changes will be
+# lost.  Perhaps that would be fixed later - perhaps writes shouldn't be cached
+# after all.  But if write to register is really needed from TCL - then it
+# should be done via "arc jtag" for now.
+proc arc_v3_examine_target { {target ""} } {
```

Comment:
Just curious, how different that part is from other ARC processors? Maybe there's a way to re-use something from ARCv2?